log4j-1.2-api/src/main/java/org/apache/log4j/builders/BuilderManager.java

```diff
@@ -71,7 +71,7 @@ public Appender parseAppender(String name, String className, String prefix, Stri
             String filterPrefix, Properties props, PropertiesConfiguration config) {
         Objects.requireNonNull(plugins, "plugins");
         Objects.requireNonNull(className, "className");
-        PluginType<?> plugin = plugins.get(className.toLowerCase());
+        PluginType<?> plugin = plugins.get(className.toLowerCase().trim());
```

Comment:
This could cause an NPE, see the bull check on the line below.

Reply:
AFAIK `toLowerCase()` does not return nulls. :-) Besides, `className` is protected by an `Objects.requireNonNull`.

Reply:
I am sorry @ppkarwasz I misread the line, I though the trim was applied to the result of the get, my mistake.